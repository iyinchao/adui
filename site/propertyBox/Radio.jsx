import React from "react"
import styles from "./components/propertyBox.scss"
import CodeBox from "./components/CodeBox"
import { Button, Switch, Radio } from "../../components"
import { SIZES } from "./consts"

export default class PropertyBox extends React.Component {
  state = {
    disabled: false,
    size: "small",
  }

  render() {
    const { disabled, size } = this.state

    const codes = `<Radio.Group
  defaultValue={0}
  ${disabled ? "disabled" : ""}
  ${size === "small" ? "" : `size="${size}"`}
>
  <Radio value={1}>单选</Radio>
  <Radio value={2}>单选</Radio>
  <Radio value={3}>单选</Radio>
</Radio.Group>`

    return (
      <div className={styles.wrapper}>
        <div className={styles.property}>
          <div className={styles.left}>
            <Radio.Group defaultValue={1} {...this.state}>
              <Radio value={1}>单选</Radio>
              <Radio value={2}>单选</Radio>
              <Radio value={3}>单选</Radio>
            </Radio.Group>
          </div>
          <div className={styles.right}>
            <div className={styles.rightTitle}>可配置项</div>
            <dl className={styles.controls}>
              <dt className={styles.controlTitle}>尺寸</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {SIZES.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === size}
                      onClick={() => this.setState({ size: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="禁用态"
                  unCheckedText="禁用态"
                  checked={disabled}
                  onChange={(param) => this.setState({ disabled: param })}
                />
              </div>
            </dl>
          </div>
        </div>
        <CodeBox codes={codes} />
      </div>
    )
  }
}
