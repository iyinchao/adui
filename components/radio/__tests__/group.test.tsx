import * as React from "react"
import { mount } from "enzyme"
import TestRenderer from "react-test-renderer"
import Radio from ".."

const { Group } = Radio

describe("Radio", () => {
  it("Group 是否正确渲染", () => {
    const wrapper = TestRenderer.create(
      <Group size="large">
        <Radio>children0</Radio>
        <Radio>children1</Radio>
      </Group>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it("Group 是否能内部驱动", () => {
    const onChange = jest.fn()
    const wrapper0 = mount(
      <Group onChange={onChange}>
        <Radio>children0</Radio>
        <Radio>children1</Radio>
      </Group>
    )
    const wrapper1 = mount(
      <Group defaultValue="children0" onChange={onChange}>
        <Radio>children0</Radio>
        <Radio>children1</Radio>
      </Group>
    )
    wrapper0.find("label").at(0).simulate("click")
    expect(onChange).toHaveBeenCalledWith("children0")
    wrapper0.find("label").at(1).simulate("click")
    expect(onChange).toHaveBeenCalledWith("children1")

    expect(wrapper1.find("div").at(0).props()["data-value"]).toEqual(
      "children0"
    )
    wrapper1.find("label").at(1).simulate("click")
    expect(onChange).toHaveBeenCalledWith("children1")
  })

  it("Group 是否正确处理 value State", () => {
    // 确保优先判断 Prop defaultValue，如果存在默认值，则 Group 状态为此默认值
    const wrapper1 = mount(
      <Group defaultValue="children0">
        <Radio disabled>children0</Radio>
        <Radio value={27}>children1</Radio>
        <Radio checked value={3}>
          children2
        </Radio>
      </Group>
    )
    expect(wrapper1.find("div").at(0).props()["data-value"]).toEqual(
      "children0"
    )
    wrapper1.find("label").at(2).simulate("click")
    expect(wrapper1.find("div").at(0).props()["data-value"]).toEqual(3)

    // 确保最优先判断 Prop value，如果存在 Prop value，则 Group 完全交由外部控制，内部状态无效。
    const wrapper2 = mount(
      <Group defaultValue={123} value={27}>
        <Radio disabled value={123}>
          children0
        </Radio>
        <Radio value={27}>children1</Radio>
        <Radio checked value={3}>
          children2
        </Radio>
      </Group>
    )
    expect(wrapper2.find("div").at(0).props()["data-value"]).toEqual(27)
    wrapper2.find("label").at(2).simulate("click")
    expect(wrapper2.find("div").at(0).props()["data-value"]).toEqual(27)
  })
})
