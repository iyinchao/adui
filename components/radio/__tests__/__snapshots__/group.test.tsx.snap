// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Group 是否正确渲染 1`] = `
<div
  className="adui-radio-group"
>
  <label
    aria-checked={false}
    className="adui-radio-base adui-radio-large"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="radio"
  >
    <span
      className="adui-radio-indicator"
    />
    <span>
      children0
    </span>
  </label>
  <label
    aria-checked={false}
    className="adui-radio-base adui-radio-large"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="radio"
  >
    <span
      className="adui-radio-indicator"
    />
    <span>
      children1
    </span>
  </label>
</div>
`;
